cmd/skywire-cli/commands/config/update/root.go

```diff
@@ -0,0 +1,174 @@
+	Use:   "update",
+	Short: "update a config file",
+	PreRun: func(_ *cobra.Command, _ []string) {
+		var err error
+		if output, err = filepath.Abs(addOutput); err != nil {
+			logger.WithError(err).Fatal("Invalid config output.")
+		}
+	},
+	Run: func(cmd *cobra.Command, _ []string) {
+		mLog := logging.NewMasterLogger()
+		mLog.SetLevel(logrus.InfoLevel)
+		if cmd.Flags().Changed("serviceConfURL") {
+			updateEndpoints = true
+		}
+		var services visorconfig.Services
+		if updateEndpoints {
+			urlstr := []string{"http://", serviceConfURL, "/config"}
+			serviceConfURL = strings.Join(urlstr, "")
+			client := http.Client{
+				Timeout: time.Second * 2, // Timeout after 2 seconds
+			}
+			req, err := http.NewRequest(http.MethodGet, serviceConfURL, nil)
+			if err != nil {
+				mLog.Fatal(err)
+			}
+			res, err := client.Do(req)
+			if err != nil {
+				mLog.Fatal(err)
+			}
+			if res.Body != nil {
+				defer res.Body.Close() //nolint
+			}
+			body, err := ioutil.ReadAll(res.Body)
+			if err != nil {
+				mLog.Fatal(err)
+			}
+			err = json.Unmarshal(body, &services)
+			if err != nil {
+				mLog.Fatal(err)
+			}
+		}
+
+		conf, ok := visorconfig.ReadConfig(addInput)
+		if ok != nil {
+			mLog.WithError(ok).Fatal("Failed to parse config.")
+		}
+
+		/*
+			switch environment {
+			case "production":
+				visorconfig.SetDefaultProductionValues(conf)
+			case "testing":
+				visorconfig.SetDefaultTestingValues(conf)
+			default:
+				logger.Fatal("Unrecognized environment value: ", environment)
+			}
+		*/
+		switch setPublicAutoconnect {
+		case "true":
+			conf.Transport.PublicAutoconnect = true
+		case "false":
+			conf.Transport.PublicAutoconnect = false
+		case "":
+			break
+		default:
+			logger.Fatal("Unrecognized public autoconnect value: ", setPublicAutoconnect)
+		}
+
+		if minHops >= 0 {
+			conf.Routing.MinHops = uint16(minHops)
+		}
+
+		saveConfig(conf)
+	},
+}
+
+func changeAppsConfig(conf *visorconfig.V1, appName string, argName string, argValue string) {
+	apps := conf.Launcher.Apps
+	for index := range apps {
+		if apps[index].Name != appName {
+			continue
+		}
+		updated := false
+		for ind, arg := range apps[index].Args {
+			if arg == argName {
+				apps[index].Args[ind+1] = argValue
+				updated = true
+			}
+		}
+		if !updated {
+			apps[index].Args = append(apps[index].Args, argName, argValue)
+		}
+	}
+}
+
+func resetAppsConfig(conf *visorconfig.V1, appName string) {
+	apps := conf.Launcher.Apps
+	for index := range apps {
+		if apps[index].Name == appName {
+			apps[index].Args = []string{}
+		}
+	}
+}
+
+func saveConfig(conf *visorconfig.V1) {
+	// Save config to file.
+	if err := conf.Flush(); err != nil {
+		logger.WithError(err).Fatal("Failed to flush config to file.")
+	}
+
+	// Print results.
+	j, err := json.MarshalIndent(conf, "", "\t")
+	if err != nil {
+		logger.WithError(err).Fatal("An unexpected error occurred. Please contact a developer.")
+	}
+	logger.Infof("Updated file '%s' to: %s", output, j)
```

Comment:
that was the default way it was when I started on this. I've added a flag to --hide the output now but I usually just redirect the output to /dev/null in the skywire-autoconfig script to avoid it being printed to the terminal.

I think it makes it a little faster to work with to print the config by default but i can change it if that is your request.